src/depfile.rs

```diff
@@ -72,19 +72,31 @@ pub fn parse<'a>(scanner: &mut Scanner<'a>) -> ParseResult<Deps<'a>> {
 #[cfg(test)]
 mod tests {
     use super::*;
-    #[test]
-    fn test_parse() {
+
+    fn must_parse<'a>(s: &'a str) -> Deps<'a> {
         let mut scanner =
-            Scanner::new("build/browse.o: src/browse.cc src/browse.h build/browse_py.h\n\0");
-        let deps = match parse(&mut scanner) {
+            Scanner::new(s);
+        match parse(&mut scanner) {
             Err(err) => {
                 println!("{}", scanner.format_parse_error("test", err));
                 panic!("failed parse");
             }
-            Ok(d) => d,
+            Ok(d) => return d,
```

Comment:
You might consider making this just 'd' and drop the semi on L85, that is normally the Rust way.